### PR TITLE
New tutorial section on how to use resources

### PR DESCRIPTION
**Pull request by tacastillo**
## Summary & Motivation

Resurrecting after I goofed up #14566 and didn't want to resolve that rebase.

New tutorial section on resources. Applied [most] of the feedback.

### CODE REVIEW

File: docs/content/tutorial/connecting-to-external-services.mdx [hunk 1]
```diff
@@ -0,0 +1,184 @@
+---
+title: "Tutorial, part eight: Connecting to external services | Dagster Docs"
+description: Learn how to connect to external services in Dagster with resources
+---
+
```
Comment (erinkcochran87) [inline docs/content/tutorial/connecting-to-external-services.mdx:5]:
```suggestion
---

Tutorial, part 8: Connecting to external services

```

File: docs/content/tutorial/connecting-to-external-services.mdx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+- Standardize how connections are used across assets
+- Create a single source to monitor connections in the Dagster UI
+
+---
+
+## Step 1: Setting up a resource
+
+Suppose your Hacker News pipeline in Dagster has picked up momentum, and your stakeholders want to learn more about who is using it. You are tasked with analyzing how many people sign up for Hacker News.
+
+In the scaffolded Dagster project you made in Part 2 of the tutorial, you'll have seen a directory called `resources` with some files in it. The most important file is the `resources.py` file that exposes a resource called `DataGeneratorResource`. This resource generates simulated data about Hacker News signups. You'll use this resource to get the data needed to produce an asset for analysis.
```
Comment (erinkcochran87) [inline docs/content/tutorial/connecting-to-external-services.mdx:38]: > Part 2 of the tutorial

🤔 Add a link here?

> you'll have seen a directory called `resources` with some files in it.

❓ Thoughts on rewording this a little? `you'll have seen` feels a little odd to me - what if I didn't? What about `you may have noticed`?

File: docs/content/tutorial/connecting-to-external-services.mdx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+		        "Earliest Signup": signups["registered_at"].min(),
+		        "Latest Signup": signups["registered_at"].max(),
+		}
+	)
+```
+
+Verify this new asset works by reloading your definitions and materializing the `signups` asset. If you navigate to the `hackernews_api`'s resource page from earlier, you’ll notice its **Uses** value now says `2`: one for the asset using it and another for the job using this asset. Click on the number and ensure that your `signups` asset uses the `hackernews_api` resource.
+
+<Image
+alt="indicating how many times a resource is used in the Dagster UI"
```
Comment (erinkcochran87) [inline docs/content/tutorial/connecting-to-external-services.mdx:115]:
```suggestion
alt="Indicating how many times a resource is used in the Dagster UI"
```

File: docs/content/tutorial/connecting-to-external-services.mdx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+	resources={
+		"hackernews_api": DataGeneratorResource(num_days=365)
+	}
+)
+```
+
+Confirm this worked by materializing the `signups` asset again. The `Earliest Signup` date should be a year ago from the `Latest Signup` date.
+
+<Image
+alt="the Dagster asset graph with metadata about when the earliest signup was"
```
Comment (erinkcochran87) [inline docs/content/tutorial/connecting-to-external-services.mdx:141]:
```suggestion
alt="The Dagster asset graph with metadata about when the earliest signup was"
```

File: docs/content/tutorial/connecting-to-external-services.mdx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+			num_days=EnvVar.int("HACKERNEWS_NUM_DAYS_WINDOW"),
+		)
+	}
+)
+```
+
+Restart your Dagster instance, materialize your signups asset, and observe how the `Earliest Signup` metadata value is 30 days ago from today.
+
+# Next Steps
+
```
Comment (erinkcochran87) [inline docs/content/tutorial/connecting-to-external-services.mdx:179]:
```suggestion
---

# Next steps

```

**Review by erinkcochran87 — COMMENTED** [review]
Found a few small things - back to you!

File: docs/content/tutorial/connecting-to-external-services.mdx (hunk 3 above)
Comment (tacastillo) [reply docs/content/tutorial/connecting-to-external-services.mdx:115]: oh, u meant like _really_ small

**Review by erinkcochran87 — COMMENTED** [review]
close!

I also realized that this new section wasn't added to `/docs/content/tutorial.mdx` yet - could you add it there pls?

**Review by erinkcochran87 — COMMENTED** [review]
Could you take care of this guy? That should be it! https://github.com/dagster-io/dagster/pull/14655#pullrequestreview-1468483679

**Review by erinkcochran87 — APPROVED** [review]

**Pull request merged**